server.py

```diff
@@ -232,3 +232,29 @@ def semesterIDType(reqstr):
 
   # Convert Officer objects back to dictionaries before returning
   return jsonify([x.__dict__ for x in officers])
+
+@app.route('/api/projects', methods=['GET'])
+def getProjects():
+    class Projects():
+      """
+      Represents Officer objects from the json file
+      Used by json.load() - see 'object_hook' in the Python docs
+      """
+      def __init__(self, d):
+        # d is the dictionary passed by json.load()
+        self.name = str(d['name'])
+        self.status = d['status']
+        self.description = str(d['description'])
+        self.smallImage = d['smallImage']
+        self.largeImage = d['largeImage']
+        self.wikiLink = d['wikiLink']
+        self.githubLink = d['githubLink']
+        self.startDate = d['startDate']
+        self.projectLead = str(d['projectLead'])
+
+    # Get projects from json file
+    with open('projects.json') as projects_file:
+      projects = json.load(projects_file, object_hook=Projects)
+
+    # Convert project objects back to dictionaries before returning
+    return jsonify([x.__dict__ for x in projects])
```

Comment:
It rubs me the wrong way a little here that we're loading a file, parsing it as json into a list of dictionaries, using each dictionary to construct a different object, and then finally converting all of them back into the same json plaintext we had from the file in the first place. The only reason we'd need this class is if we needed to do some kind of filtering or logic here on the backend to reduce what we send over the wire, like we do in the Officers route, and we only added the class there once we put in that filtering logic. This whole function could literally be:
```python
with open('projects.json') as projects_file:
  return projects_file.read()
```
I don't know what frontend's filtering needs are, but I think we should either keep this function simple, or add in something that justifies the complexity.

Reply:
I just copied how you did it in the officers, I'll change it to this. The front end does the filtering, we're just sending them the whole long list. (I feel really dumb rn lol)